lib/vector-core/src/tls/settings.rs

```diff
@@ -335,6 +345,14 @@ impl TlsSettings {
 
     pub fn apply_connect_configuration(&self, connection: &mut ConnectConfiguration) {
         connection.set_verify_hostname(self.verify_hostname);
+        if let Some(server_name) = &self.server_name {
+            // Prevent native TLS lib from inferring default SNI using domain name from url.
+            connection.set_use_server_name_indication(false);
+            match connection.set_hostname(server_name) {
+                Ok(_) => (),
+                Err(e) => error!("Failed to set server name indication: {}", e),
```

Comment:
Could we bubble this error up? Given it happens at start time, I think Vector should abort if it happens.

Reply:
updated to bubble up the error.